Source/Core/Gen8/Generators/EggGenerator8.cpp

```diff
@@ -72,25 +72,31 @@ std::vector<EggState> EggGenerator8::generate(u64 seed0, u64 seed1) const
             {
                 // gen.next(2);
                 // Handle display result later
-                gen.next();
-            }
-
-            if (genderRatio == 255)
-            {
-                state.setGender(2);
-            }
-            else if (genderRatio == 254)
-            {
-                state.setGender(1);
-            }
-            else if (genderRatio == 0)
-            {
-                state.setGender(0);
+                state.setGender(gen.next(2));
+                if (genderRatio != 255 && genderRatio != 254 && genderRatio != 0)
+                {
+                    gen.next();
+                }
```

Comment:
Why is this part necessary?

Reply:
Based on previous logic, the next rand call was still affected based on the gender ratio. What would the expected gender ratio be in the filters for nidoran volbeat? A 100% one/genderless or 50/50?

Reply:
Is it accurate to what the game does?

Reply:
Will need to double check. It's more we should be seeing if the game treats them as 50/50 and has an extra gender call or treats them as their base 100% and only has the one.